keras_cv/losses/giou_loss.py

```diff
@@ -0,0 +1,172 @@
+            [KerasCV bounding box documentation](https://keras.io/api/keras_cv/bounding_box/formats/).
+        axis: the axis along which to mean the ious. Passing the string "no_reduction" implies
+            mean across no axes. Defaults to -1.
+
+    References:
+        - [GIoU paper](https://arxiv.org/pdf/1902.09630)
+        - [TFAddons Implementation](https://www.tensorflow.org/addons/api_docs/python/tfa/losses/GIoULoss)
+
+    Sample Usage:
+    ```python
+    y_true = tf.random.uniform((5, 10, 5), minval=0, maxval=10, dtype=tf.dtypes.int32)
+    y_pred = tf.random.uniform((5, 10, 4), minval=0, maxval=10, dtype=tf.dtypes.int32)
+    loss = GIoULoss(bounding_box_format = "xyWH")
+    loss(y_true, y_pred).numpy()
+    ```
+    Usage with the `compile()` API:
+    ```python
+    model.compile(optimizer='adam', loss=keras_cv.losses.GIoULoss())
+    ```
+    """
+
+    def __init__(self, bounding_box_format, axis=-1, **kwargs):
+        super().__init__(**kwargs)
+        self.bounding_box_format = bounding_box_format
+        self.axis = axis
+
+    def _compute_enclosure(self, boxes1, boxes2):
+        y_min1, x_min1, y_max1, x_max1 = tf.split(
+            boxes1[..., :4], num_or_size_splits=4, axis=-1
+        )
+        y_min2, x_min2, y_max2, x_max2 = tf.split(
+            boxes2[..., :4], num_or_size_splits=4, axis=-1
+        )
+        boxes2_rank = len(boxes2.shape)
+        perm = [1, 0] if boxes2_rank == 2 else [0, 2, 1]
+        # [N, M] or [batch_size, N, M]
+        zeros_t = tf.cast(0, boxes1.dtype)
+
+        enclose_ymin = tf.minimum(y_min1, tf.transpose(y_min2, perm))
+        enclose_xmin = tf.minimum(x_min1, tf.transpose(x_min2, perm))
+        enclose_ymax = tf.maximum(y_max1, tf.transpose(y_max2, perm))
+        enclose_xmax = tf.maximum(x_max1, tf.transpose(x_max2, perm))
+        enclose_width = tf.maximum(zeros_t, enclose_xmax - enclose_xmin)
+        enclose_height = tf.maximum(zeros_t, enclose_ymax - enclose_ymin)
+        enclose_area = enclose_width * enclose_height
+
+        return enclose_area
+
+    def _compute_giou(self, boxes1, boxes2):
+        boxes1_rank = len(boxes1.shape)
+        boxes2_rank = len(boxes2.shape)
+
+        if boxes1_rank not in [2, 3]:
+            raise ValueError(
+                "compute_iou() expects boxes1 to be batched, or "
+                f"to be unbatched. Received len(boxes1.shape)={boxes1_rank}, "
+                f"len(boxes2.shape)={boxes2_rank}. Expected either len(boxes1.shape)=2 AND "
+                "or len(boxes1.shape)=3."
+            )
+        if boxes2_rank not in [2, 3]:
+            raise ValueError(
+                "compute_iou() expects boxes2 to be batched, or "
+                f"to be unbatched. Received len(boxes1.shape)={boxes1_rank}, "
+                f"len(boxes2.shape)={boxes2_rank}. Expected either len(boxes2.shape)=2 AND "
+                "or len(boxes2.shape)=3."
+            )
+
+        target = bounding_box.preserve_rel(
```

Comment:
If it's converted to rel_yxyx, or rel_yxhw, don't we need to provide `image_shape` during convert_format so it can support ALL input box format?
In any case not converting to relative format sounds more clear to me

Reply:
NVM, this can be resolved